apps/api-gateway/src/public-data/public-data.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { PostCategoryMessagePattern } from '@shared/constants/message-pattern';
+
+@Injectable()
+export class PublicDataService {
+  public constructor(
+    @Inject('PUBLIC_DATA_SERVICE') private readonly publicDataClient: ClientProxy,
+  ) {}
+
+  public getPostData() {
+    const pattern = PostCategoryMessagePattern.GET_POST_DATA;
+    return this.publicDataClient.send(pattern, {});
```

Comment:
이렇게 표현하는 거랑은 뉘앙스가 어떻게 다를까요?
```suggestion
    return this.publicDataClient.send(PostCategoryMessagePattern.GET_POST_DATA, {});
```

Reply:
해당 부분은 의도적으로 팀원들이 api-gateway와
MicroService 와의 MessagePattern 방식으로 하는 소통에 대해
어떤 값들이 전달되어야 하는지 가시적으로 보여주기 위해
일부러 작성하긴 했습니다!
근데, 써주신 코드의 방식이 좀 더 적은 코드로 표현 가능할 것 같아 수정 하는 방향으로 
진행 하겠습니다!